src/firefly/js/templates/lightcurve/LcPhaseFoldingPopup.jsx

```diff
@@ -170,18 +224,193 @@ function PhaseFoldingSetting() {
+            <div style={{...childStyle}}>
+                <PhaseFoldingChart height={childStyle.height} width={width}/>
+            </div>
         </div>
     );
 }
 
 /**
+ * @summary calculate phase and return as text
+ * @param {number} time
+ * @param {number} timeZero
+ * @param {number} period
+ * @param {number} dec
+ * @returns {string}  folded phase (positive or negative) with 'dec'  decimal digits
+ */
+function getPhase(time, timeZero, period,  dec=DEC) {
+    var q = (time - timeZero)/period;
+    var p = q >= 0  ? (q - Math.floor(q)) : (q + Math.floor(-q));
+
+    return p.toFixed(dec);
+}
+
+/**
+ * @summary get flux series in terms of folded phase
+ * @param {Object} fields
+ * @returns {*}
+ */
+function getPhaseFlux(fields) {
+    var rawTable = getTblById(RAW_TABLE);
+    var {columns, data} = rawTable.tableData;    // column names and data
+
+    var {timeCol, period, tzero, flux} = fields;
+    const tc = timeCol ? timeCol.value : '';
+    const fc = flux ? flux.value: '';
+    const tIdx = tc ? columns.findIndex((c) => (c.name === tc)) : -1;  // find time column
+    const fIdx = fc ? columns.findIndex((c) => (c.name === fc)) : -1;  // find flux column
+
+    if (tIdx < 0 || fIdx < 0) return {};
+
+    var pd, tz;
+
+    if (period){
+        if (!period.valid) return {};    // invalid period
+        pd = parseFloat(period.value);
+    } else {
+        pd = periodRange.min;
+    }
+    if (tzero) {
+        if (!tzero.valid) return {};      // invalid period
+        tz = parseFloat(tzero.value);
+    } else {
+        tz = timeZero;
+    }
+
+    var phaseV = data.map((d) => {
+        return parseFloat(getPhase(d[tIdx], tz, pd));
+    });
+
+    var minPhase = Math.min(...phaseV);
+    var maxPhase = Math.max(...phaseV) + 1.0;
+
+    var xy = data.reduce((prev, d, idx) => {
+        var phase1, phase2, fluxV;
+
+        fluxV = parseFloat(d[fIdx]);
+        phase1 = phaseV[idx];
+        phase2 = phase1 + 1.0;
+        prev.push([phase1, fluxV], [phase2, fluxV]);
+
+        return prev;
+    }, []);
+
+    return Object.assign({data: xy}, {minPhase, maxPhase, period: pd.toFixed(DEC), flux: fc});
+}
+
+/**
+ * @summary 2D xyplot component on phase folding panel
+ */
+class PhaseFoldingChart extends Component {
+    constructor(props) {
+        super(props);
+
+        var fields = FieldGroupUtils.getGroupFields(pfkey);
+        var {width, height, showTooltip=true} = props;
+        var {data, minPhase = 0.0, period, flux} = getPhaseFlux(fields);
+
+        this.state = {
+            fields,
+            config: {
+                chart: {
+                    type: 'scatter',
+                    borderColor: '#a5a5a5',
+                    borderWidth: 1,
+                    borderRadius: 5,
+                    zoomType: 'xy',
+                    height,
+                    width
+                },
+                title: {
+                    text: 'Flux vs. Phase'
```

Comment:
This is just my opinion: I think this title does not add any new info. Also, title and legend together take too much space from the chart. I would put Period=...  in the title, and remove legend.